src/Graph.cpp

```diff
@@ -153,10 +152,26 @@ void Graph::dijkstra(const string &source, const string &destination) {
     unordered_map<string, Station*> unvisited;
     unordered_map<string, Station*> previous;
     unordered_map<string, int> distance;
+    int sf = 1;
+    int df = 1;
+
 
     for(auto &station: stations){
         distance[station.first] = INT_MAX;
         unvisited[station.first] = station.second;
+
+        if(station.first == source) sf = 0;
+        if(station.first == destination) df = 0;
+    }
+
+    if(sf || df){
+        if(sf){
+            cout << "Invalid Source Station Name.\n";
+        }
+        if(df){
+            cout << "Invalid Destination Station Name.\n";
+        }
+        return;
     }
```

Comment:
Programming principles:

```suggestion
    if(sf){
        cout << "Invalid Source Station Name.\n";
        return;
    }
    if(df){
        cout << "Invalid Destination Station Name.\n";
        return;
    }
```